### ISSUE TITLE
Allow Scrollbar Auto Creation on All Elements

### ISSUE THREAD

**Issue by Snayff** [MyreMylar/pygame_gui#75]
**Is your feature request related to a problem? Please describe.**
The vertical scroll bar creates itself for the textbox when the text is too large - this works very well but is missing on other ui elements, perhaps most importantly on the ui_window.

**Describe the solution you'd like**
Have both vertical and horizontal scroll bars auto create for all ui_elements, as with the text_box. Allow parameter to enable or disable this.

**Comment by MyreMylar** [comment]
_Fairly_ sure we don't want this for **all** UI elements. I think a button or a scroll bar with another scroll bar on it would look pretty odd.

However, it is likely needed for windows in some form, I'll probably do this after I've added horizontal scroll bars and refactored the windows code again (it's just undergone some heavy surgery in the 0.5.0 branch).

I think after I've made the UIPanel element to separate out some functionality, I can probably pull more stuff into the basic window class (like menu bars and close buttons) and make it optional in there as part of the theming.

This might be a 0.7.0 thing after I go over the text code in 0.6.0. I'll schedule it there for now.

**Comment by Snayff** [comment]
Haha, agreed, buttons with scroll bars would be quite odd. 

Having close buttons on windows by default/option sounds like a great idea. What about tabs? I imagine they're pretty much just buttons that replace one window with another but there might be some mileage in having a pre-baked option.

**Comment by Snayff** [comment]
I tried porting from the textbox to my class (inheriting from window) but the more I dig into it the more I realise it's more work than a lift and shift. It looks like the window doesnt have some of the required info. I am not surprised that the textbox and window are dissimilar, and you are likely already aware, but just in case I wanted to feedback.

**Comment by MyreMylar** [comment]
> What about tabs? I imagine they're pretty much just buttons that replace one window with another but there might be some mileage in having a pre-baked option.

Yeah, I've thought about doing 'proper' tabs for windows, the main thing I'm waiting on a bit there is the eventual proper release of Pygame 2 as I have hopes it'll allow me to speed up and simplify my existing drawing code and make things like the rounded corners more viable to have everywhere (if you haven't noticed if you add a bunch of rounded corners to everything in a complicated UI thing it eventually can take about half a second to draw things which feels unresponsive). With rounded corners on a tab you want a surprisingly tricky shape to dynamically draw where the 'exposed' edge is rounded in the normal fashion and the 'connecting' edge is has curves that go the other way, all of which means it will take longer to put together on the fly.

Anyway, it's something I'm interested in longer term as I think they are a fairly useful thing to have for UI heavy games - though of course you can probably fake it pretty well already with some carefully placed buttons.

**Comment by MyreMylar** [comment]
Does the UIScrollableContainer in 0.5.5 help solve this? You can see how I use it in pygame paint.

**Comment by Snayff** [comment]
In principle, absolutely. Is it significantly different from the UIPanel, outside of the scrollbars? 

I've also rediscovered our conversation about tabs. I'll pull it out to a separate issue.
